Bottom: cfbda9f29d2b48b11d2d3de02bd00fc79bf09091
Top:    9e7708cdd935686631aa76ead1b2156cd07c94f5
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-29 19:50:06 +0100

Refresh of hid-gtk-extend-layer-selector-

---

diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index 639e2be..a3fcb19 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -384,19 +384,26 @@ GtkWidget *
 ghid_layer_selector_new (void)
 {
   int i;
-  GtkCellRenderer *renderer1 = ghid_cell_renderer_visibility_new ();
-  GtkCellRenderer *renderer2 = gtk_cell_renderer_text_new ();
-  GtkTreeViewColumn *opacity_col =
-      gtk_tree_view_column_new_with_attributes ("", renderer1,
-                                                "active", VISIBLE_COL,
-                                                "color", COLOR_COL, NULL);
-  GtkTreeViewColumn *name_col =
-      gtk_tree_view_column_new_with_attributes ("", renderer2,
-                                                "text", TEXT_COL,
-                                                "font", FONT_COL,
-                                                NULL);
-
-  GHidLayerSelector *ls = g_object_new (GHID_LAYER_SELECTOR_TYPE, NULL);
+  GtkCellRenderer *renderer1;
+  GtkCellRenderer *renderer2;
+  GtkTreeViewColumn *opacity_col;
+  GtkTreeViewColumn *name_col;
+  GHidLayerSelector *ls;
+
+  renderer1 = ghid_cell_renderer_visibility_new ();
+  renderer2 = gtk_cell_renderer_text_new ();
+  g_object_set (renderer2, "editable-set", TRUE, NULL);
+
+  opacity_col = gtk_tree_view_column_new_with_attributes ("", renderer1,
+                                                          "active", VISIBLE_COL,
+                                                          "color", COLOR_COL, NULL);
+  name_col = gtk_tree_view_column_new_with_attributes ("", renderer2,
+                                                       "text", TEXT_COL,
+                                                       "font", FONT_COL,
+                                                       "editable", EDITABLE_COL,
+                                                       NULL);
+
+  ls = g_object_new (GHID_LAYER_SELECTOR_TYPE, NULL);
 
   /* action index, active, color, text, font, is_separator */
   ls->list_store = gtk_list_store_new (N_COLS, G_TYPE_POINTER, G_TYPE_INT,
